### DISCUSSION TITLE
How to log the best validation metrics

### DISCUSSION THREAD

**Discussion by FDUguchunhui** [Lightning-AI/pytorch-lightning#20564] [General]
I am trying to use Lightning for hyperparameter tuning, so I would like to have validation metric for each hyperparameter combination. This is easy when I evaluate test dataset with the final model for each hyperparameter combination since I can ignore test metric performance and directly compare their validation metrics. However, when I want to avoid epoches as a hyperparameter and only use save the best model, the logged validation metrics is not synchronized with the model used for testing (the logged validation is at the model after the end of N epoch but model used for testing could be M < N epoches). How could also log this best validation loss into tensorboard or other logger? 

Thank you!

```
checkpoint_callback = ModelCheckpoint(monitor='val_loss', dirpath=cfg.other.model_path, filename='best_model', save_top_k=1, mode='min')
early_stop_callback = EarlyStopping(monitor='val_loss', patience=cfg.train.early_stop_patience, mode='min')


trainer = pl.Trainer(max_epochs=cfg.train.num_epochs,
                     callbacks=[checkpoint_callback, early_stop_callback],
                     accelerator='cpu',
                     logger=logger,
                     # fast_dev_run=True
                     )
trainer.fit(model, datamodule=dm)
trainer.test(model, datamodule=dm, ckpt_path="best")
```
